keps/sig-scheduling/2458-node-resource-score-strategy/README.md

```diff
@@ -0,0 +1,674 @@
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+The motivation is two fold:
+
+1) Reduce the complexity of configuring the default scheduler:
+
+Configuring the scheduler plugins is a tedious task. The relatively large number of plugins
+that the default scheduler supports doesn't make that task easier: more plugins means larger
+number of combinations of enabled/disabled plugins.
+
+Moreover, some combinations don't make sense and potentially harmful. For example, specific
+to this KEP, enabling both least and most allocated scoring plugins is not useful.
+
+Finally, we are planning to add more resource fit scoring strategies, like Best/WrostFit which
+allows preferring nodes with the least/most amount of available absolute resources that can host the
+pod; also, Least/MostAllocatable which allows preferring nodes with the least/most amount of 
+allocatable resources that can host the pod. Adding those strategies as separate plugins will
+make the scheduler configuration problem even worse.
+
+2) A step towards allowing workloads to express their resource fit scoring strategy via pod spec.
+
+Similar to how we allow workloads to express spread and affinity preferences, we want to allow
+pods to express node resource fit preferences. This is important to achieve higher utilization
+when running a mix of serving and batch workloads on the same cluster. We believe that the
+changes we make here will enable this long term vision.
+
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+- Allow users to configure resource fit preferences using a single plugin configuration
+- Deprecate resource-based scoring plugins that implement individual strategies, specifically: 
+  - `NodeResourcesLeastAllocated`
+  - `NodeResourcesMostAllocated`
+  - `RequestedToCapacityRatio`
+
```

Comment:
How about adding one bullet here, like "ensure the design of this plugin extensible to be able to accommodate flexible fit strategies in the future"?

Reply:
done.